packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,195 @@
+
+      const workerFiles = WorkerAssets.listWorkerFiles(distServerPath);
+      for await (const workerFile of workerFiles) {
+        yield [`server/${workerFile.normalizedPath}`, workerFile.data];
+      }
+    }
+
+    async function uploadTarballAsync(tarPath: string): Promise<any> {
+      const uploadUrl = await getSignedDeploymentUrlAsync(graphqlClient, exp, {
+        appId: projectId,
+      });
+
+      const { response } = await uploadAsync({
+        url: uploadUrl,
+        filePath: tarPath,
+        compress: false,
+        headers: {
+          accept: 'application/json',
+        },
+      });
+      if (response.status === 413) {
+        throw new Error(
+          'Upload failed! (Payload too large)\n' +
+            `The gzipped files in "dist/server/" (at: ${distServerPath}) exceed the maximum file size.`
+        );
+      } else if (!response.ok) {
+        throw new Error(`Upload failed! (${response.statusText})`);
+      } else {
+        const json = await response.json();
+        if (!json.success || !json.result || typeof json.result !== 'object') {
+          throw new Error(json.message ? `Upload failed: ${json.message}` : 'Upload failed!');
+        }
+        return json.result;
+      }
+    }
+
+    async function uploadAssetsAsync(
+      assetMap: WorkerAssets.AssetMap,
+      uploads: Record<string, string>
+    ): Promise<void> {
+      if (typeof uploads !== 'object' || !uploads) {
+        return;
+      }
+
+      // TODO(@kitten): Batch and upload multiple files in parallel
+      const uploadParams: UploadParams[] = [];
+      for await (const asset of WorkerAssets.listAssetMapFiles(distClientPath, assetMap)) {
+        const uploadURL = uploads[asset.normalizedPath];
+        if (uploadURL) {
+          uploadParams.push({ url: uploadURL, filePath: asset.path });
+        }
+      }
+
+      const progress = {
+        total: uploadParams.length,
+        pending: 0,
+        percent: 0,
+        transferred: 0,
+      };
+
+      const updateProgress = createProgressTracker({
+        total: progress.total,
+        message(ratio) {
+          const percent = `${Math.floor(ratio * 100)}`;
+          const details = chalk.dim(
+            `(${progress.pending} Pending, ${progress.transferred} Completed, ${progress.total} Total)`
+          );
+          return `Uploading client assets: ${percent.padStart(3)}% ${details}`;
+        },
+        completedMessage: 'Uploaded client assets for worker deployment',
+      });
+
+      try {
+        for await (const signal of batchUploadAsync(uploadParams)) {
+          if ('response' in signal) {
+            progress.pending--;
+            progress.percent = ++progress.transferred / progress.total;
+          } else {
+            progress.pending++;
+          }
+          updateProgress({ progress });
+        }
+      } catch (error: any) {
+        updateProgress({ isComplete: true, error });
+        Log.error(error);
+        return;
```

Comment:
```suggestion
      } catch (error: any) {
        updateProgress({ isComplete: true, error });
        Log.error(error);
        throw error;
```

Shouldn't this error be thrown?

Reply:
I always got double errors here when throwing, so it's either `Log.error` and not getting the stack trace, or `throw error` and getting the stack trace but not both 🤔 I'm a little confused on this tbh, and am not sure which is better, but happy to switch to just throwing in this particular process